File: backends/python-fastapi/app/domain/invitation/entity.py
"""
招待ドメイン エンティティモジュール。

Author: Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
"""
from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass
class Invitation:
    """招待のドメインエンティティ（SQLAlchemy タグなし）。"""
    id: int = 0
    token: str = ""
    role: int = 2
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

File: backends/ts-hono/src/domain/invitation/entity.ts
/**
 * 招待ドメイン エンティティモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** 招待のドメインエンティティ。 */
export interface Invitation {
  id: number;
  token: string;
  role: number;
  createdAt: Date | null;
  updatedAt: Date | null;
}
